marketplacetests/in_app_payments/in_app.py

```diff
@@ -2,29 +2,39 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from marionette.by import By
+from marionette import expected, By, Wait
 from gaiatest.apps.base import Base
 
 
 class InAppPayment(Base):
 
     # Products
-    _buy_product_button_locator = (By.CSS_SELECTOR, '.item > button')
     _bought_product_locator = (By.CSS_SELECTOR, '#bought .item > h4')
+    _buy_product_button_locator = (By.XPATH, "//h4[text()='%s']/following-sibling::button")
+    _server_select_locator = (By.ID, 'server')
 
-    def __init__(self, marionette):
+    def __init__(self, marionette, server):
         Base.__init__(self, marionette)
         self.apps.switch_to_displayed_app()
-
-    def tap_buy_product(self):
-        self.wait_for_element_displayed(*self._buy_product_button_locator)
-        self.marionette.find_element(*self._buy_product_button_locator).tap()
-        from marketplacetests.firefox_accounts.app import FirefoxAccounts
-        return FirefoxAccounts(self.marionette)
+        self.set_server('API: %s' % server)
```

Comment:
Currently all of the values in the select are prefixed with `API: `, so it seemed clearer in the test to just specify the server name, rather than the whole string `API: marketplace-dev.allizom.org`. I can see reasons to do either. If you think it's better to specify the whole string in the caller I can make that change.